Makefile

```diff
@@ -23,6 +23,14 @@ in: inplace # just a shortcut
 inplace:
 	$(PYTHON) setup.py build_ext -i
 
+dev-meson:
+	# Temporary script to try the experimental meson build. Once meson is
+	# accepted as the default build tool, this will go away.
+	python build_tools/build-meson-editable-install.py
+
+clean-meson:
+	pip uninstall -y scikit-learn
```

Comment:
Good poin, I would keep it like this for now. My main goal with `make clean-meson` is to be able to switch back to work with setuptools easily and possibly back again with Meson. Keeping Meson build artifacts is useful for the latter.

`make clean` will remove the `build` folder which contain Meson build artifacts (unless you make the effort to pass a different build directory with a complicated command like `pip install -e . -Csetup-args=...`)

Reply:
Meson, unlike setuptools, is very safe here. It does a very good job at marking artifacts as stale and rebuilding them when needed. You don't generally need to manually clean it, because you can just re-run `ninja` and keep the good artifacts and regenerate the outdated artifacts.